src/objects/js/components/admin/permissions/permission-form.js

```diff
@@ -0,0 +1,91 @@
+import React, { useState } from "react";
+
+import { CheckboxInput, TextInput, SelectInput } from "../../forms/inputs";
+import { authFields } from "./auth-fields";
+
+
+const PermissionForm = (props) => {
+    const {objectFields, dataFieldChoices, tokenChoices, objecttypeChoices, modeChoices, formData} = props;
+
+    const {values, errors} = formData;
+    const [useFields, setUseFields] = useState(values["use_fields"]);
+    const [fields, setFields] = useState(values["fields"] ? values["fields"].split(",") : []);
+    const [dataFields, setDataFields] = useState(dataFieldChoices[values["object_type"]]);
+
+    return (
+     <fieldset className="module aligned">
+        <div className="form-row">
+            <SelectInput
+                name="token_auth"
+                id="id_token_auth"
+                label="Token auth:"
+                choices={tokenChoices}
+                initialValue={values["token_auth"]}
+                errors={errors["token_auth"]}
+            />
+        </div>
+
+        <div className="form-row">
+            <SelectInput
+                name="object_type"
+                id="id_object_type"
+                label="Object type:"
+                choices={objecttypeChoices}
+                initialValue={values["object_type"]}
+                errors={errors["object_type"]}
+                onChange={(value) => {
+                    setDataFields(dataFieldChoices[value]);
+                    // remove other data fields from selected
+                    const newFields = fields.filter(x => !x.startsWith("record__data__"))
+                    setFields(newFields);
+                }}
+            />
+        </div>
+
+        <div className="form-row">
+            <SelectInput
+                name="mode"
+                id="id_mode"
+                label="Mode:"
+                choices={modeChoices}
+                initialValue={values["mode"]}
+                errors={errors["mode"]}
+            />
+
+        </div>
+
+        <div className="form-row">
+            <CheckboxInput
+                name="use_fields"
+                id="id_use_fields"
+                label="Use fields"
```

Comment:
I think it may be a bit more clear if the label was `Use field-based authorization` instead of `Use fields`, or if there was help text to explain this field a bit more